packages/optimizer/lib/fetchRuntimeParameters.js

```diff
@@ -15,7 +15,7 @@
  */
 'mode strict';
 
-const URL_BENTO_COMPONENT_INFO = 'https://amp.dev/static/bento-components.json';
+const URL_COMPONENT_VERSIONS = 'https://amp.dev/static/files/component-versions.json';
```

Comment:
Does this correspond to the `latest` version of each component?

Reply:
It seems some extensions are missing, namely:


1. `amp-action-macro`
1. `amp-base-carousel` 
1. `amp-cache-url` (this lacks a `latest` in [the spec](https://github.com/ampproject/amphtml/blob/main/extensions/amp-cache-url/validator-amp-cache-url.protoascii))
1. `amp-google-assistant-assistjs`
1. `amp-mega-menu`
1. `amp-story-panning-media`
1. `amp-stream-gallery`
1. `amp-truncate-text`

Reply:
Also, it doesn't seem accurate. It has this:

```
"amp-lightbox-gallery": "1.0",
```

And yet 1.0 is not even a [valid version in the spec](https://github.com/ampproject/amphtml/blob/e3abe630d3e0bcce78115e0b51d5da776e427904/extensions/amp-lightbox-gallery/validator-amp-lightbox-gallery.protoascii#L22-L23).

Reply:
The invalid latest version appears to be an issue with:

1. `amp-facebook-comments` (has 1.0 but [should be 0.1](https://github.com/ampproject/amphtml/blob/e3abe630d3e0bcce78115e0b51d5da776e427904/extensions/amp-facebook-comments/validator-amp-facebook-comments.protoascii#L22-L23))
2. `amp-lightbox-gallery` (has 1.0 but [should be 0.1](https://github.com/ampproject/amphtml/blob/e3abe630d3e0bcce78115e0b51d5da776e427904/extensions/amp-lightbox-gallery/validator-amp-lightbox-gallery.protoascii#L22-L23))
3. `amp-twitter` (has 1.0 but [should be 0.1](https://github.com/ampproject/amphtml/blob/e3abe630d3e0bcce78115e0b51d5da776e427904/extensions/amp-twitter/validator-amp-twitter.protoascii#L22-L23))

Reply:
Also, it seems `bento-components.json` is not completely. Namely, it is missing these Bento components:

* [`amp-video-iframe`](https://github.com/ampproject/amphtml/blob/main/extensions/amp-video-iframe/1.0/amp-video-iframe.js) (v1.0)
* [`amp-vimeo`](https://github.com/ampproject/amphtml/blob/main/extensions/amp-vimeo/1.0/amp-vimeo.js) (v1.0)

Reply:
@westonruter Drive-by comment to suggest an alternative to https://amp.dev/static/bento-components.json:
- The source of truth for Bento components is in [this file](https://github.com/ampproject/amphtml/blob/main/build-system/compile/bundles.config.extensions.json) (search for `"npm": true`).
- These are the components that will be automatically published to npm (see [design doc](https://docs.google.com/document/d/1hUmJl-NkpujkJyVpRRARTeoXHSiSf3VvMm5ujv5D10A/edit)).
- In case it helps, we have a wrapper [node script](https://github.com/ampproject/amphtml/blob/main/build-system/npm-publish/get-extensions.js) that can extract the raw list of components.

/cc @estherkim @jridgewell @kristoferbaxter for visibility / additional comment.

Reply:
@rsimha this is perfect! Exactly what I was looking for. Am I right, that `latestVerstion` will always point to the latest valid prod version?

Background: Bento components have been causing the problem here, but the real problem is that it's not possible to determine experimental components from the validation rules.

Reply:
> Am I right, that latestVerstion will always point to the latest valid prod version?

Yep!

Reply:
Yes! Git tags are created as part of the nightly cuts.